app/controllers/spree/admin/reports_controller_decorator.rb

```diff
@@ -250,23 +215,53 @@ def xero_invoices
     @distributors = Enterprise.is_distributor.managed_by(spree_current_user)
     @order_cycles = OrderCycle.active_or_complete.accessible_by(spree_current_user).order('orders_close_at DESC')
 
-    @report = OpenFoodNetwork::XeroInvoicesReport.new spree_current_user, params
+    @report = OpenFoodNetwork::XeroInvoicesReport.new spree_current_user, params, render_content?
     render_report(@report.header, @report.table, params[:csv], "xero_invoices_#{timestamp}.csv")
   end
 
+  private
+
+  # Some actions are changing the `params` object. That is unfortunate Spree
+  # behavior and we are building on it. So we have to look at `params` early
+  # to check if we are searching or just displaying a report search form.
+  def cache_search_state
+    search_keys = [
+      # search parameter for ransack
+      :q,
+      # common in all reports, only set for CSV rendering
+      :csv,
+      # `button` is included in all forms. It's not important for searching,
+      # but the Users & Enterprises report doesn't have any other parameter
+      # for an empty search. So we use this one to display data.
+      :button,
```

Comment:
Could we just use the presence of a `:button` param as a proxy for a search request for all reports? That would seem to negate the need to check all of these other params as well...?